packages/dds/tree/src/simple-tree/treeApi.ts

```diff
@@ -141,11 +156,58 @@ export const nodeApi: TreeApi = {
 
 /**
  * A collection of events that can be raised by a {@link TreeNode}.
+ *
+ * @privateRemarks
+ * TODO: add a way to subscribe to a specific field (for afterShallowChange and afterDeepChange).
+ * Probably have object node and map node specific APIs for this.
+ *
+ * TODO: ensure that subscription API for fields aligns with API for subscribing to the root.
+ *
+ * TODO: add more wider area (avoid needing tons of afterShallowChange registration) events for use-cases other than afterDeepChange.
+ * Some ideas:
+ *
+ * - afterDeepChange, but with some subtrees/fields/paths excluded
+ * - helper to batch several afterShallowChange calls to a afterDeepChange scope
+ * - parent change (ex: registration on the parent field for a specific index: maybe allow it for a range. Ex: node event takes optional field and optional index range?)
+ * - new content inserted into subtree. Either provide event for this and/or enough info to afterDeepChange to find and search the new sub-trees.
+ * Add separate (non event related) API to efficiently scan tree for given set of types (using low level cursor and schema based filtering)
+ * to allow efficiently searching for new content (and initial content) of a given type.
+ *
  * @public
  */
 export interface TreeNodeEvents {
```

Comment:
All these events currently could be logically applied to parts of a tree other than just Nodes, so maybe these should be renamed "TreeChangeEvents", and we can reuse this type for applying to fields, the root, and even field subranges.

Reply:
The rename seems ok to me. Thinking of worst case scenarios: we make the change, and we realize that events for fields/root have particularities that require a separate event; TreeChangeEvents would still sound like an ok place to add them. If we keep TreeNodeEvents and discover particularities for field/root events we are kind of forced to add a separate interface and `&` them somewhere, or rename it as a breaking change. So the rename would provide a bit of future-proofing, I think. I'll do it as part of taking over this proposal/PR.